app/org/sagebionetworks/bridge/services/AuthenticationService.java

```diff
@@ -171,6 +171,10 @@ public UserSession signIn(Study study, CriteriaContext context, SignIn signIn) t
         
         Account account = accountDao.authenticate(study, signIn);
 
+        if (intentService.registerIntentToParticipate(study, account)) {
+            account = accountDao.getAccount(AccountId.forId(study.getIdentifier(), account.getId()));    
+        }
```

Comment:
It looks like this is non-trivial and it will be executed on every sign-in, which could be a lot. Would it be better to move this to verify channel, which is most likely only executed once?

(Similarly with channelSignIn())

Reply:
Hmm I hadn't thought of that. Yes that's a good idea. I'll look at moving it to the verification step. We just need to be sure we verify through that method if people sign in without explicitly verifying.